ui/hooks/usePolledTransactions.ts

```diff
@@ -0,0 +1,354 @@
+  transactionGroup: {
+    nonce: number;
+    initialTransaction: any;
+    primaryTransaction: any;
+    transactions: any[];
+  };
+  timeMs: number;
+  id: string;
+};
+
+const groupTransactionsByDate = (
+  transactionGroups: any[],
+  getTransactionTimestamp: (item: any) => number,
+  shouldSort = true,
+) => {
+  const groupedTransactions: any[] = [];
+
+  if (!transactionGroups) {
+    return groupedTransactions;
+  }
+
+  transactionGroups.forEach((transactionGroup) => {
+    const timestamp = getTransactionTimestamp(transactionGroup);
+    const date = formatDateWithYearContext(timestamp, 'MMM d, y', 'MMM d');
+
+    const existingGroup = groupedTransactions.find(
+      (group) => group.date === date,
+    );
+
+    if (existingGroup) {
+      existingGroup.transactionGroups.push(transactionGroup);
+      if (shouldSort) {
+        // Sort transactions within the group by timestamp (newest first)
+        existingGroup.transactionGroups.sort((a: any, b: any) => {
+          const aTime = getTransactionTimestamp(a);
+          const bTime = getTransactionTimestamp(b);
+          return bTime - aTime; // Descending order (newest first)
+        });
+      }
+    } else {
+      groupedTransactions.push({
+        date,
+        dateMillis: timestamp,
+        transactionGroups: [transactionGroup],
+      });
+    }
+    if (shouldSort) {
+      // Sort date groups by timestamp (newest first)
+      groupedTransactions.sort((a, b) => b.dateMillis - a.dateMillis);
+    }
+  });
+
+  return groupedTransactions;
+};
+
+// Group EVM transaction groups, non‑EVM transactions, or unified items by date alike
+const groupAnyTransactionsByDate = (items: any[]) =>
+  groupTransactionsByDate(
+    items,
+    (item) => {
+      // Prefer precomputed timeMs (unified items)
+      if (typeof item?.timeMs === 'number') {
+        return item.timeMs;
+      }
+      // EVM transactionGroup
+      if (item?.primaryTransaction?.time) {
+        return item.primaryTransaction.time;
+      }
+      // Non‑EVM transaction
+      if (item?.timestamp) {
+        return item.timestamp * 1000;
+      }
+      return 0;
+    },
+    true,
+  );
+
+const fetchTransactions = async (
+  address: string,
+  enabledNetworks: any,
+): Promise<ApiResponse> => {
+  // Build the networks parameter from already filtered enabled networks
+  const networkChainIds = Object.keys(enabledNetworks?.eip155 || {});
+  const networksParam = networkChainIds.join(',');
+
+  // Build URL with search parameters, using unencoded networks value
+  const url = new URL(
+    `https://accounts.api.cx.metamask.io/v1/accounts/${address}/transactions`,
+  );
+  // Manually append networks parameter to avoid encoding commas
+  url.search = `?networks=${networksParam}&sortDirection=DESC`;
+
+  const response = await fetch(url.toString());
+  if (!response.ok) {
+    throw new Error('Failed to fetch transactions');
+  }
+  return response.json();
+};
+
+const transformTransactionsToUnified = (
```

Comment:
All these mapping will be moved

Can be further simplified we can fetch only the necessary fields we need from the metamask API endpoint